src/app/settings/components/project-role-edit-panel/project-role-edit-panel.tsx

```diff
@@ -50,37 +59,18 @@ export class ProjectRoleEditPanel extends React.Component<ProjectRoleEditPanelPr
                         </div>
                         <h4>Policies:</h4>
                         <FormField formApi={api} label='' field='policies' component={TextArea}/>
-                        <h4>JWT Tokens:</h4>
-                        { api.values.jwtTokens !== null && api.values.jwtTokens.length > 0 ? (
-                            <div className='argo-table-list'>
-                                <div className='argo-table-list__head'>
-                                    <div className='row'>
-                                        <div className='columns small-3'>ID</div>
-                                        <div className='columns small-3'>ISSUED AT</div>
-                                        <div className='columns small-3'>EXPIRES AT</div>
-                                    </div>
-                                </div>
-                                {api.values.jwtTokens.map((jwtToken: models.JwtToken) => (
-                                    <div className='argo-table-list__row' key={`${jwtToken.iat}`}>
-                                        <div className='row'>
-                                            <div className='columns small-3'>
-                                                {jwtToken.iat}
-                                            </div>
-                                            <div className='columns small-3'>
-                                                {new Date(jwtToken.iat * 1000).toDateString()}
-                                            </div>
-                                            <div className='columns small-3'>
-                                                {jwtToken.exp == null ? 'None' : new Date(jwtToken.exp * 1000).toDateString()}
-                                            </div>
-                                        </div>
-                                    </div>
-                                ))}
-                            </div>
-                        ) : <div className='white-box'><p>Role has no JWT tokens</p></div> }
+                        <ProjectRoleJWTTokens tokens={api.values.jwtTokens} api={api}/>
```

Comment:
Don't render `ProjectRoleJWTTokens` if role is not created since creation will fail

Reply:
fixed